DESCRIPTION: This patch adds a check, restricting Node
registration in the cluster while it has no internal IP.
~~~
Index: pkg/cloudprovider/vsphere/nodemanager.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pkg/cloudprovider/vsphere/nodemanager.go b/pkg/cloudprovider/vsphere/nodemanager.go
--- a/pkg/cloudprovider/vsphere/nodemanager.go	(revision 2e4610dbef94d74ca6b27a3cf8a84b069be8e148)
+++ b/pkg/cloudprovider/vsphere/nodemanager.go	(revision 00b8010f9773c666fae000315b436a89b09f54c3)
@@ -424,6 +424,17 @@
 		return fmt.Errorf("unable to find suitable IP address for node %s with IP family %s", nodeID, ipFamily)
 	}

+	var internalAddressExists bool
+	for _, addr := range addrs {
+		if addr.Type == v1.NodeInternalIP {
+			internalAddressExists = true
+			break
+		}
+	}
+	if !internalAddressExists {
+		return fmt.Errorf("no Internal IP found in the node %s addresses:\n%+v", nodeID, addrs)
+	}
+
 	klog.V(2).Infof("Found node %s as vm=%+v in vc=%s and datacenter=%s",
 		nodeID, vmDI.VM, vmDI.VcServer, vmDI.DataCenter.Name())
 	klog.V(2).Info("Hostname: ", oVM.Guest.HostName, " UUID: ", oVM.Summary.Config.Uuid)
